livemaps-api/ServiceFunctions/FeatureStatePublisher.cs

```diff
@@ -50,26 +47,23 @@ public static class FeatureStatePublisher
 
             if (prerequisites)
             {
+                //Create a new container if the container not exists.
                 await container.CreateIfNotExistsAsync();
-                var bacmapRef = container.GetBlockBlobReference(recentDataFile);
-
-                IEnumerable<dynamic> recentdata;
-                using (var ms = new MemoryStream())
-                {
-                    await bacmapRef.DownloadToStreamAsync(ms);
-                    ms.Position = 0;
-                    using (StreamReader reader = new StreamReader(ms, Encoding.UTF8))
-                    {
-                        var bacmapstr = reader.ReadToEnd();
-                        recentdata = JsonConvert.DeserializeObject<IEnumerable<dynamic>>(bacmapstr);
-                    }
-                }
+                //Create a Blob client
+                var bacmapRef = container.GetBlobClient(recentDataFile);
+                //Downloads a blob from the service.
+                BlobDownloadResult result = await bacmapRef.DownloadContentAsync();
+                string deviceStateData = result.Content.ToString();
+                IEnumerable<dynamic> recentData;
+                recentData = JsonConvert.DeserializeObject<IEnumerable<dynamic>>(deviceStateData);
 
                 foreach (EventData eventData in events)
                 {
                     try
                     {
-                        string messageBody = Encoding.UTF8.GetString(eventData.Body.Array, eventData.Body.Offset, eventData.Body.Count);
+                        //Convert EventBody Type to string.
+                        string messageBody = eventData.EventBody.ToString();
+                        //Deserialize the EventBody data to TagObject class.
                         var dataItems = JsonConvert.DeserializeObject<IEnumerable<TagObject>>(messageBody);
```

Comment:
This should be updated to use `System.Text.Json`

Reply:
@jsquire I've updated to use System.Text.Json, please review.